server/app/services/mongodb/migrator.rb

```diff
@@ -64,17 +67,27 @@ def migrations
 
     def migrate
       ensure_indexes
-      with_dlock('mongodb_migrate', 60) do
+      release_stale_lock
+      with_dlock(LOCK_NAME, LOCK_TIMEOUT) do
         migrate_without_lock
       end
     end
 
+    def release_stale_lock
```

Comment:
Or better yet... drop the entire "cleanup" thing, and replace it with something like

```ruby
with_dlock('mongodb_migrate', timeout: 60, expires: 3 * 60) do
    ...
end
```

Advantage: the lock expiry is right there in the same place where the lock is created so it's much clearer to understand how it will behave, as opposed to having the cleanup in some different place.